### PR TITLE
Dev refactor domain label extractor to use base processor test case

### PR DESCRIPTION
**Pull request by urdit52**
Add specific and generic rules

### CODE REVIEW

**Comment by ekneg54** [comment]
Please resolve linting issues and unused imports:
* test_domain_label_extractor.py
* processor.py
*

File: logprep/processor/domain_label_extractor/rule.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def __init__(self, filter_rule: FilterExpression, domain_label_extractor_cfg: di
         self._target_field = domain_label_extractor_cfg["target_field"]
         self._output_field = domain_label_extractor_cfg["output_field"]
 
-    def __eq__(self, other: 'DomainLabelExtractorRule') -> bool:
+    def __eq__(self, other: "DomainLabelExtractorRule") -> bool:
         return (other.filter == self._filter) and (self._compare_set == other.compare_set)
```
Comment (ekneg54) [inline logprep/processor/domain_label_extractor/rule.py:48]: my editor says, that there is no `_compare_set` attribute on self. And as I see, there is no test for `__eq__` method in `rule.py`

File: tests/unit/processor/domain_label_extractor/test_domain_label_extractor_rule.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import copy
```
Comment (ekneg54) [inline tests/unit/processor/domain_label_extractor/test_domain_label_extractor_rule.py:1]: please add tests for equality checks

File: logprep/processor/domain_label_extractor/factory.py [hunk 3]
```diff
@@ -15,15 +18,16 @@ def create(name: str, configuration: dict, logger) -> DomainLabelExtractor:
...
-                                                      configuration.get('tld_lists', None),
-                                                      configuration.get('tagging_field_name', 'tags'),
-                                                      logger)
-        domain_label_extractor.add_rules_from_directory(configuration['rules'])
-
-        return domain_label_extractor
+        return DomainLabelExtractor(
+            name=name,
+            configuration=configuration,
+            tld_lists=configuration.get("tld_lists", None),
```
Comment (ekneg54) [inline logprep/processor/domain_label_extractor/factory.py:24]: the `configuration` is part of the parameter list. there is no need to extract parts of it as seperate parameters. It is possibly the right time to refactor the `__init__` method of `DomainLabelExtractor` ;)

File: logprep/processor/domain_label_extractor/processor.py [hunk 4]
```diff
@@ -9,37 +9,52 @@
...
+        message += " ".join(skipped_fields)
 
         super().__init__(name, message)
 
 
 class DomainLabelExtractor(RuleBasedProcessor):
     """Splits a domain into it's parts/labels."""
 
-    def __init__(self, name: str, tree_config: str, tld_lists: list, tagging_field_name: str, logger: Logger):
+    def __init__(
```
Comment (ekneg54) [inline logprep/processor/domain_label_extractor/processor.py:50]: there should only 3 parameters: `name`, `configuration`, `logger`

File: logprep/processor/domain_label_extractor/factory.py [hunk 5]
```diff
@@ -2,11 +2,14 @@
 
 from logprep.processor.base.factory import BaseFactory
 from logprep.processor.domain_label_extractor.processor import DomainLabelExtractor
+from logprep.processor.processor_factory_error import InvalidConfigurationError
 
 
 class DomainLabelExtractorFactory(BaseFactory):
     """Factory used to instantiate DomainLabelExtractor processors."""
 
+    mandatory_fields = ["type", "generic_rules", "specific_rules", "tree_config"]
```
Comment (ekneg54) [inline logprep/processor/domain_label_extractor/factory.py:11]: what is with the `tld_lists` ... are they optional?

File: tests/unit/processor/domain_label_extractor/test_domain_label_extractor_rule.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    DomainLabelExtractorRule,
+    InvalidDomainLabelExtractorDefinition,
+)
+from logprep.processor.domain_label_extractor.factory import DomainLabelExtractorFactory
+from logprep.processor.domain_label_extractor.processor import (
+    DomainLabelExtractor,
+    DuplicationError,
+)
+
+logger = getLogger()
```
Comment (ekneg54) [inline tests/unit/processor/domain_label_extractor/test_domain_label_extractor_rule.py:23]: logger isn't used anywhere

File: tests/unit/processor/domain_label_extractor/test_domain_label_extractor_rule.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+rel_tld_list_path = "tests/testdata/external/public_suffix_list.dat"
+tld_list = f"file://{Path().absolute().joinpath(rel_tld_list_path).as_posix()}"
```
Comment (ekneg54) [inline tests/unit/processor/domain_label_extractor/test_domain_label_extractor_rule.py:25]: tld_list isn't used anywhere

**Review by ekneg54 — CHANGES_REQUESTED** [review]
please fix linting in tests and production code files.
for tests it is ok, to resolve by disabling some issues on top of the test file, if to hard to fix

File: logprep/processor/domain_label_extractor/factory.py (hunk 5 above)
Comment (urdit52) [reply logprep/processor/domain_label_extractor/factory.py:11]: They are. [https://logprep.readthedocs.io/en/latest/user_manual/configuration/processor.html#domain-label-extractor](Logprep Docs)

File: logprep/processor/domain_label_extractor/rule.py (hunk 1 above)
Comment (urdit52) [reply logprep/processor/domain_label_extractor/rule.py]: fixed.

**Review by ekneg54 — APPROVED** [review]
reviewed together

**Review by ppcad — APPROVED** [review]

**Pull request merged**